app/frontend/Calendar/DayView/WeekView.svelte

```diff
@@ -83,28 +83,24 @@
   }
 
   let days: Date[] = [];
+  let visibleEvents: Collection<Event>;
+  let allDayEvents: Collection<Event>;
   $: start, setDays();
   function setDays() {
     let startTime = new Date(start);
     if (showDays > 3) {
       startTime.setDate(startTime.getDate() - 1);
     }
-    startTime.setHours(startHour);
-    startTime.setMinutes(0);
-    startTime.setSeconds(0);
-    startTime.setMilliseconds(0);
+    let filterStart = startTime.setHours(startHour, 0, 0, 0);
     days = [];
     for (let i = 0; i < showDays; i++) {
       days.push(new Date(startTime));
-      startTime.setDate(startTime.getDate() + 1)
+      startTime.setDate(startTime.getDate() + 1);
     }
-  }
-
-  let allDayEvents: Collection<Event>;
-  $: start, $events, setAllDayEvents();
```

Comment:
The trigger conditions for `setDays()` and `setAllDayEvents()` are different.
* Why would you re-calculate `days` when `$events` change?
* Smaller functions are better than larger functions.

Reply:
I don't actually recalculate anything because `$events` can't actually change.

Reply:
> $events is always appGlobal.calendarEvents

Yes, but the `$` subscribes to changes, so whenever calendar events are added or removed, this triggers and the `setDays()` function runs. Given that `days` is *not* dependent on events, and changing `days` triggers a refresh of the entire UI, it should not change every time an event is added or removed.